### PR TITLE
Add Owner for DBT

### LINKED ISSUE #6801: Pick up DBT column descriptions and owners
column info should be in the catalog. Owners, we will add just if they exist in OM. Otherwise we'll just log them down

### PR DESCRIPTION
**Pull request by OnkarVO7**
### Describe your changes :

Fix #6801 

#
### Type of change :

- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java [hunk 1]
```diff
@@ -618,12 +618,20 @@ public Table deleteCustomMetric(UUID tableId, String columnName, String metricNa
...
     // Carry forward the table description from the model to table entity, if empty
     if (nullOrEmpty(table.getDescription())) {
       table.setDescription(dataModel.getDescription());
     }
+
+    // Carry forward the table owner from the model to table entity, if empty
+    if (table.getOwner() == null) {
+      table.setOwner(dataModel.getOwner());
+      updateOwner(table, oldOwner, table.getOwner());
+    }
```
Comment (nahuelverdugo) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java:633]:
```suggestion
    if (table.getOwner() == null) {
      storeOwner(table, dataModel.getOwner());
    }
```
And we can remove line 621.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java [hunk 2]
```diff
@@ -618,12 +618,20 @@ public Table deleteCustomMetric(UUID tableId, String columnName, String metricNa
   @Transaction
   public Table addDataModel(UUID tableId, DataModel dataModel) throws IOException {
     Table table = dao.findEntityById(tableId);
+    EntityReference oldOwner = table.getOwner();
```
Comment (nahuelverdugo) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java:621]:
```suggestion
```

**Review by nahuelverdugo — APPROVED** [review]
LGTM but I miss some unit tests at least. I also left an optional suggestion.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java (hunk 1 above)
Comment (OnkarVO7) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java:633]: Yes making this change thanks 👍

**Review by nahuelverdugo — APPROVED** [review]

**Pull request merged**